github4s/src/test/scala/github4s/integration/ReposSpec.scala

```diff
@@ -313,6 +313,57 @@ trait ReposSpec extends BaseIntegrationSpec {
     response.statusCode shouldBe notFoundStatusCode
   }
 
+  "Repos >> UserIsCollaborator" should "return true when the user is a collaborator" taggedAs Integration in {
+    val response = clientResource
+      .use { client =>
+        Github[IO](client, accessToken).repos
+          .userIsCollaborator(
+            validRepoOwner,
+            validRepoName,
+            validUsername,
+            headers = headerUserAgent
+          )
+      }
+      .unsafeRunSync()
+
+    testIsRight[Boolean](response, r => r should be(true))
+    response.statusCode shouldBe noContentStatusCode
+  }
+
+  it should "return false when the user is not a collaborator" taggedAs Integration in {
+    val response = clientResource
+      .use { client =>
+        Github[IO](client, accessToken).repos
+          .userIsCollaborator(
+            validRepoName,
+            validRepoName,
+            invalidUsername,
+            headers = headerUserAgent
+          )
+      }
+      .unsafeRunSync()
+
+    testIsRight[Boolean](response, r => r should be(false))
+    response.statusCode shouldBe notFoundStatusCode
+  }
+
+  it should "return error when other errors occur" taggedAs Integration in {
+    val response = clientResource
+      .use { client =>
+        Github[IO](client, accessToken).repos
+          .userIsCollaborator(
+            validRepoName,
```

Comment:
If we are trying to trigger an UnauthorizedError here, maybe use strings that makes that more obvious? `"different-owner"` and `"some-repo"` or something like that.

Reply:
I don't think it will be easy to generate one as gh tends to just hide things behind 404s.

Reply:
Hmm do you think it's worth removing this test then?

Reply:
could use an invalid access token?